File: back/src/main/java/com/julien/plop/Language.java
package com.julien.plop;

import java.util.Locale;

public enum Language {
    FR, EN;

    public static Language valueOfSafe(String language) {
        try {
            return Language.valueOf(language.toUpperCase());
        } catch (IllegalArgumentException e) {
            return FR;
        }
    }
}

File: mobile/lib/generic/config/language.dart

import 'dart:ui';

enum Language {
  en,
  fr;

 static Language byDefault() {
    return Language.fr;
  }

  Locale toLocale() {
    switch(this) {
      case Language.en:
        return const Locale('en');
      case Language.fr:
        return const Locale('fr');
    }
  }

}